### ISSUE TITLE
Native support for Steam Controller

### ISSUE THREAD

**Issue by cm-t** [recalbox/recalbox-os#466]
Hi,

I've recently received my **Steam Controller**, after pluged it on the raspi2 using recalbox, the controller is not well reconized, and when trying to use an other controller to get inside the mapping survey to configure the Steam controller, **some inputs are not detected**, means I can't finnish the setup wizard. I am then stucked on the wizard. Some keys already mapped + impossible to map some key,  get stuck on "Y" button step.

Note : _I don't know about the native support of steam controller on the linux kernel natively, but on Ubuntu 14.04 LTS, without any system modification, I must run the Steam Client to use it (I still beleive the steam client mostly binds the keys than anything more)_.

**Comment by rockaddicted** [comment]
Hello, are you on the last 3.3.0 beta ?
Can you look your recalbox.conf file, and try to switch input driver ?

```
## Set retroarch input driver (auto, udev, sdl2)
## If you don't have issues with your controllers, let auto
global.inputdriver=auto
```

More information about the recalbox.conf file here : https://github.com/digitalLumberjack/recalbox-os/wiki/recalbox.conf-%28EN%29

**Comment by cm-t** [comment]
@rockaddicted I beleive I am on v3.3.0-beta-9 (fresh install, not by upgrades)

Currently looking at that file (…)

**Comment by cm-t** [comment]
Well, I tryed auto, udev, with reboot : 

When steam controller is wireless connected with its dongle, gamepad's buttons are not really detected (can't wizard setup).

When I use the steam controller wiht an usb link, it got detected, with auto/udev, and it detected as a keyboard (as seen on the setup wizard asking to long press a button to know what gamepad to configure, here it display "keyboard" few seconde before starting the button keypress survey.
I still get stuck on the X/Y step, like if no input are sent to system, (… I don"t know what's happending technicly, but I can send you log or whatever you need)

**Comment by cm-t** [comment]
Hey, 

So, for now, using the wireless dongle, steamcontroller does not works, and with usb link, only some keys work (through it might works with a decent config that i didn't find).

Just a fresh news about steamcontroller :

 * https://github.com/ynsta/steamcontroller

>I plan to make it work on platform where steam is not working (ARM linux boards for example).

I beleive it is definitly something to watch :eyes:

**Comment by torch2424** [comment]
Retropie (4.1) just announced support for the steam controller, using: https://github.com/ynsta/steamcontroller

Perhaps we can implement in recalbox?

I'll see if I can get this installation working on my current pi 3

**Comment by substring** [comment]
but non one among the devs has such pad.

To test it, you need to clone the recalbox-buildroot project, add the required libraries (i'm afraid you'd need to create a package for the enum34 python package), build recalbox, and test ...

**Comment by synaestheory** [comment]
@substring I've got a steam controller, when you're looking for that archive, are you just looking to have someone plug their steam controller in and then run that shell script? Or are you suggesting that they run the build and bundle the additional libraries?

**Comment by substring** [comment]
@synaestheory well ... both :D

**Comment by bpollman** [comment]
Hey just jumping in on the fun. Started using RecalBox a couple weeks ago and am familiar with buildroot so I thought I would add in support for the steam controller.  

I had to add in python-libusb1 as well as enabled enum34.

https://github.com/recalbox/recalbox-buildroot/pull/577

**Comment by mcronce** [comment]
In case anybody's interested in having this feature on 4.0 and doesn't mind building from source, https://github.com/mcronce/recalbox-buildroot/tree/rb-4.0.X-with-steamcontroller - I forked the rb-4.0.X branch and cherry-picked the commits from @bpollman's PR

My copy's still building, but the merge conflicts were pretty simple and the changes pretty self-contained all things considered, so I'm pretty confident.  Will post back here with results tomorrowish once I've had a chance to test it.

**Comment by webartoli** [comment]
I'm not familiar with buildroot. How can use steam controller on rb ?

**Comment by bpollman** [comment]
to enable it, just set "controllers.steam.enabled=1" in the configuration

**Comment by webartoli** [comment]
On a clean rb 4.0.2 it doesn't work. I've tried to add the `controllers.steam.enabled=1` row, and I have also tried to disable other kind of controllers to avoid conflicts, but it doesn't work.

I've also checked out my controller to exclude hardware issues and it works like a charm on both PC and  RetroPie distro

**Comment by rockaddicted** [comment]
has been merged on recalbox 4.1 version.
so you have to wait until work on updating infrastructure is complete before you can upgrade.

**Comment by substring** [comment]
solved with https://github.com/recalbox/recalbox-os/releases/tag/4.1.0
